### PR TITLE
Update to tslint 4

### PR DESCRIPTION
**Pull request by DorianGrey**
- For full changelog, see: https://github.com/palantir/tslint/blob/master/CHANGELOG.md
- Some rules have been removed, since the TS-compiler can already handle them (most of them since 1.8)
- Once this gets merged and published, we should sharpen the template's compiler options accordingly

@svi3c

### CODE REVIEW

File: src/scripts.ts [hunk 1]
```diff
@@ -47,14 +46,17 @@ export class ScriptLinter extends EventEmitter {
...
-    const lintOptions = {configuration: this.configuration, formatter: "verbose"};
+    const lintOptions: ILinterOptions = {fix: false, formatter: "verbose"};
 
     return Promise.all(files.map(file =>
       utils.readFile(file).then(content => ({file, content}))))
       .then((fileObjs) =>
         new Promise((resolve, reject) => {
           for (let fileObj of fileObjs) {
-            const lintingResult = new Linter(fileObj.file, fileObj.content, lintOptions).lint();
+            const linter = new Linter(lintOptions);
```
Comment (sv2dev) [inline src/scripts.ts:56]: Now that we don't have to bind the linter to one file with the constructor, we could instantiate it only once, right?

File: src/scripts.ts [hunk 2]
```diff
@@ -2,18 +2,17 @@ import * as _ from "lodash";
...
+import {ILinterOptions, LintResult, Linter} from "tslint";
 import * as karma from "karma";
 
 import {ScriptsLintingConfig, ScriptsTestConfig} from "./config/config.interface";
 import * as utils from "./utils";
 
 export class ScriptLinter extends EventEmitter {
 
-  private configuration = JSON.parse(fs.readFileSync("tslint.json", "utf8"));
+  private configuration                               = JSON.parse(fs.readFileSync("tslint.json", "utf8"));
```
Comment (sv2dev) [inline src/scripts.ts]: Those spacings are inconsistent with the rest of this project. I would vote for not having them.

**Review by sv2dev — CHANGES_REQUESTED** [review]

File: src/scripts.ts (hunk 2 above)
Comment (DorianGrey) [reply src/scripts.ts]: Oh, yes - accidentially used auto-format here (align multiple lines of declarations and assignments is part of the config I'm using by default).

=> Done

File: src/scripts.ts (hunk 1 above)
Comment (DorianGrey) [reply src/scripts.ts:56]: Hm... it's possible to instantiate the linter only once, yes. However, if we do so, we get a single linter result for **all** linted files, which makes it more complicated to re-map the result to properly adjust our `lintingErrors` object. The code would roughly look like:
```typescript
const linter = new Linter(lintOptions);
for (let fileObj of fileObjs) {
  linter.lint(fileObj.file, fileObj.content, this.configuration);
}
const lintingResult = linter.getResult();
const perFile = _.groupBy(lintingResult.failures, (failure: RuleFailure) => failure.getFileName());
_.each(perFile, (failures: RuleFailure[], fileName: string) => {
  if (failures.length === 0) {
    delete this.lintingErrors[fileName];
  } else {
    this.lintingErrors[fileName] = failures;
  }
});
```
The problem is that we have to group `lintingResult.failures` after retrieving all failures, but files that do not contain any more failures are no longer on that list.

To properly deal with this, it would be required to also add entries from `lintingErrors` to `perFile`, and handle them as empty arrays...

Long story short: I'd suggest using a single instance of the linter makes the code more complicated than it is atm., while only having a minor potential to save runtime performance (I'd even guess it would worse).

Comment (sv2dev) [reply src/scripts.ts:56]: All right :)

**Review by sv2dev — APPROVED** [review]
